tool/doc.dart

```diff
@@ -628,6 +629,21 @@ class RuleHtmlGenerator {
     return 'Dart SDK: $sdkVersion • <small>(Linter $linterVersion)</small>';
   }
 
+  State get state => rule.state;
+
+  String get stateString {
+    if (state.isDeprecated) {
+      return '<span style="color:orangered;font-weight:bold;" >${state.label}</span>';
+    }
+    if (state.isRemoved) {
```

Comment:
Totally your call, but I prefer using an `else if` structure when only one branch can be taken.